File: sdk/android/sample/src/main/kotlin/app/tryskilly/sdk/PolicyAndRealtimeExample.kt
package app.tryskilly.sdk

// This sample assumes the generated UniFFI Kotlin files from sdk/android/generated
// are added to your Android module source set.
import uniffi.skilly_core_mobile_sdk.*

object PolicyAndRealtimeExample {
    fun runDemo() {
        val policyInput = MobilePolicyInput(
            userId = "android-user-123",
            entitlementState = MobileEntitlementState.TRIAL,
            trialSecondsUsed = 400UL,
            usageSecondsUsed = 0UL,
            adminWorkosUserIds = emptyList()
        )

        val policyDecision = canStartTurnForMobile(policyInput)
        println("Allowed=${policyDecision.allowed}, reason=${policyDecision.reason}")

        val events = listOf(
            MobileRealtimeEvent(MobileRealtimeEventType.TURN_STARTED, "turn-android-1", null),
            MobileRealtimeEvent(MobileRealtimeEventType.AUDIO_CAPTURE_COMMITTED, "turn-android-1", null),
            MobileRealtimeEvent(MobileRealtimeEventType.RESPONSE_STARTED, "turn-android-1", null),
            MobileRealtimeEvent(MobileRealtimeEventType.RESPONSE_COMPLETED, "turn-android-1", null)
        )

        val summary = replayRealtimeEventsForMobile(events)
        if (summary != null) {
            println("Phase=${summary.phaseName}, turnsCompleted=${summary.turnsCompleted}")
        } else {
            println("Replay failed")
        }
    }
}

File: sdk/ios/sample/PolicyAndRealtimeExample.swift
import Foundation

// This sample assumes the generated UniFFI Swift files from sdk/ios/generated
// are included in your target.
func runPolicyAndRealtimeDemo() {
    let policyInput = MobilePolicyInput(
        userId: "ios-user-123",
        entitlementState: .trial,
        trialSecondsUsed: 400,
        usageSecondsUsed: 0,
        adminWorkosUserIds: []
    )

    let policyDecision = canStartTurnForMobile(input: policyInput)
    print("Allowed: \(policyDecision.allowed), reason: \(String(describing: policyDecision.reason))")

    let events = [
        MobileRealtimeEvent(eventType: .turnStarted, turnId: "turn-ios-1", message: nil),
        MobileRealtimeEvent(eventType: .audioCaptureCommitted, turnId: "turn-ios-1", message: nil),
        MobileRealtimeEvent(eventType: .responseStarted, turnId: "turn-ios-1", message: nil),
        MobileRealtimeEvent(eventType: .responseCompleted, turnId: "turn-ios-1", message: nil),
    ]

    if let summary = replayRealtimeEventsForMobile(events: events) {
        print("Phase: \(summary.phaseName), turns completed: \(summary.turnsCompleted)")
    } else {
        print("Replay failed.")
    }
}
